common/src/main/java/org/apache/rocketmq/common/compression/CompressorFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.rocketmq.common.compression;
+
+public class CompressorFactory {
+
+    private static final Lz4Compressor LZ4_COMPRESSOR = new Lz4Compressor();
+    private static final ZlibCompressor ZLIB_COMPRESSOR = new ZlibCompressor();
+    private static final ZstdCompressor ZSTD_COMPRESSOR = new ZstdCompressor();
+
```

Comment:
You should use a Compressor[] to hold these different Compressors, and assign an id number to each Compressor, and finally assign one of them as default.

Reply:
> finally assign one of them as default.

I think init of compressType has finish this default assign. And may be I shoule use a enummap to store compressor.

Reply:
good idea